sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,272 @@
+from sympy.physics.mechanics import (Body, Lagrangian, KanesMethod, LagrangesMethod,
+                                    RigidBody, Particle)
+from sympy.physics.mechanics.method import _Methods
+
+__all__ = ['JointsMethod']
+
+
+class JointsMethod(_Methods):
+    """Method for formulating the equations of motion using a set of interconnected bodies with joints.
+
+    Parameters
+    ==========
+
+    newtonion : Body or ReferenceFrame
+        The newtonion(inertial) frame.
+    *joints : Joint
```

Comment:
Check to make sure coordinates and speeds of every joint are unique from each other or raise an error.

Reply:
Added!